gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -798,99 +804,26 @@ case class ColumnarOverrideRules(session: SparkSession)
-    countFallbacks(plan) >= queryFallbackThreshold
-  }
-
-  private def countFallbacks(plan: SparkPlan): Int = {
-    var fallbacks = 0
-    def countFallback(plan: SparkPlan): Unit = {
-      plan match {
-        // Another stage.
-        case _: QueryStageExec =>
-          return
-        case ColumnarToRowExec(p: GlutenPlan) =>
-          logDebug(s"c2r: $p")
-          fallbacks = fallbacks + 1
-        // Possible fallback for leaf node.
-        case leafPlan: LeafExecNode if !leafPlan.isInstanceOf[GlutenPlan] =>
-          fallbacks = fallbacks + 1
-        case _ =>
-      }
-      plan.children.foreach(p => countFallback(p))
-    }
-    countFallback(plan)
-    fallbacks
-  }
-
-  /**
-   * Ported from ApplyColumnarRulesAndInsertTransitions of Spark. Inserts an transition to columnar
-   * formatted data.
-   */
-  private def insertRowToColumnar(plan: SparkPlan): SparkPlan = {
-    if (!plan.supportsColumnar) {
-      // The tree feels kind of backwards
-      // Columnar Processing will start here, so transition from row to columnar
-      RowToColumnarExec(insertTransitions(plan, outputsColumnar = false))
-    } else if (!plan.isInstanceOf[RowToColumnarTransition]) {
-      plan.withNewChildren(plan.children.map(insertRowToColumnar))
-    } else {
-      plan
-    }
-  }
-
-  /**
-   * Ported from ApplyColumnarRulesAndInsertTransitions of Spark. Inserts RowToColumnarExecs and
-   * ColumnarToRowExecs where needed.
-   */
-  def insertTransitions(plan: SparkPlan, outputsColumnar: Boolean): SparkPlan = {
-    if (outputsColumnar) {
-      insertRowToColumnar(plan)
-    } else if (plan.supportsColumnar) {
-      // `outputsColumnar` is false but the plan outputs columnar format, so add a
-      // to-row transition here.
-      ColumnarToRowExec(insertRowToColumnar(plan))
-    } else if (!plan.isInstanceOf[ColumnarToRowTransition]) {
-      plan.withNewChildren(plan.children.map(insertTransitions(_, outputsColumnar = false)))
-    } else {
-      plan
-    }
-  }
-
-  // Just for test use.
-  def enableAdaptiveContext(): Unit = {
-    isAdaptiveContext = true
-  }
-
   override def postColumnarTransitions: Rule[SparkPlan] = plan =>
     PhysicalPlanSelector.maybe(session, plan) {
-      if (fallbackPolicy == "query" && !isAdaptiveContext && fallbackWholeQuery(plan)) {
-        logWarning("Fall back to run the query due to unsupported operator!")
-        insertTransitions(originalPlan, outputsColumnar = false)
-      } else if (fallbackPolicy == "stage" && isAdaptiveContext && fallbackWholeStage(plan)) {
-        logWarning("Fall back the plan due to meeting the whole stage fallback threshold!")
-        insertTransitions(originalPlan, outputsColumnar = false)
-      } else {
-        transformPlan(postOverrides(), plan, "post")
+      val planWithFallbackPolicy = transformPlan(fallbackPolicy(), plan, "fallback")
+      val finalPlan = planWithFallbackPolicy match {
+        case FallbackNode(fallbackPlan) =>
+          fallbackPlan
```

Comment:
Would you like to add one-line comment to describe why post overrides are skipped? Thanks.

Reply:
The main issue is `postOverrides` will conver vanilla c2r to velox cr2. If we decide to fallback, we should use vanilla c2r

Reply:
addressed